Feature/updates (#74)
infinum/eightshift-docs: website/docs/basics/blocks-intro.md

You can, but you shouldn't. We use this key to provide the CSS selectors on all our custom blocks in the block editor. Additional styles are added to this selector for the wrapper to work on full width and remove some of the native block editor styles. 

> So don't remove or change this 😅.

Reviewer: Can't leave a comment on that line but there's a heading right above it:

> ### Why is this my blocks folder called custom and not blocks, for example?

I'd change that to: 

> ### Why is my blocks folder called `custom` and not `blocks`, for example?

### Simple vs Compound blocks

In a nutshell, there is no difference between blocks. The WordPress core block is a block, but we like to make a distinction between what is simple and what is a compound block.

> **Simple block** is a block that is used just as is and provides a small functionality like heading, paragraph, button, etc.

Reviewer: ```suggestion
**Simple block** is a block that is used just as is and provides a small isolated functionality. For example: heading, paragraph, button, etc.
```

> **Compound block** is a block that builds from multiple components and provides a bigger functionality.

Reviewer: ```suggestion
**Compound block** is a block built from multiple components. For example: card, featured posts, etc
```

### Naming is hard

*There are only two hard things in Computer Science: cache invalidation and naming things. - Phil Karlton*

> Yes, naming is hard, and no matter how long is your development experience you will always struggle with names for your components, block, files, variables, functions, etc.

Reviewer 2: ```suggestion
Yes, naming is hard, and no matter how long your development experience is you will always struggle with names for your components, block, files, variables, functions, etc.
```

Here are some of our recommendations to ease your pain:

> * You should never name your component based on its usage. For example, you have a component card for custom post-type books, you should never call this component `card-book` bat rather use a more generic name like `card-product`.

Reviewer: ```suggestion
* Always try to name your block based on **what they are**, rather than **what they will be used for**. For example, if you have a component card for custom post-type `books`, you should never call this component `card-book`. Instead, use a more generic name like `card-product`.
```

Reviewer 2: The `product` also seems specific, no? Maybe use `card-item`? Or something more general. Product is used if you are using some kind of web stores.

> * Make your names as generic as possible for better reusability.

Reviewer 3: ```suggestion
* Make your names as generic as possible for better reusability, but specific enough to fully understand what the block/component is used for.
```

I would suggest to add this part. Not sure if everyone agrees but for me it seems useful.

* Always think about the future. How can this feature be used in some other way?

> * When naming your attributes ask yourself: "is this attribute going to be used in any other way?". For example, you have an attribute for adding font-weight: bold to your text. You can create an attribute called font-weight and set it as a `boolean` type and that will be ok if you have only one font-weight. A better way would be to put it as a string and provide and SelectControl component if there is any possibility that in the future you will have additional font-weight.

Reviewer: ```suggestion
* When naming your attributes ask yourself: "is this attribute going to be used in any other way?". For example, you have an attribute for adding font-weight: bold to your text. You can create an attribute called font-weight and set it as a `boolean` type and that will be ok if you have only one font-weight. A better way would be to put it as a string and provide a `SelectControl` component if there is any possibility that in the future you will have additional font-weight.
```

> * Name booleans positively. Example: `isValid`, `isLoading`, `isComplete`.

Reviewer: ```suggestion
* Name booleans positively as a question. Example: `isValid`, `isLoading`, `isComplete`.
```

> * Don’t hesitate to use longer names: We use short words even if it doesn’t provide full insight about the code which is a bad thing to do.

Reviewer: > We use short words even if it doesn’t provide full insight about the code which is a bad thing to do.

Do we need to open an issue for this? :D

Reviewer 2: Yeah, kinda goes against what you are saying in this chapter 😅

Author: fixed

* Use singular names.
* The variables or functions should be named by their work: Name of variables/functions should always try to express their meaning without diving into the code base try to pack meaningful information inside the name.
* Naming should be simple enough to be understood by everyone: Using complex words to describe a simple thing only creates hassle while reading the code. Also, use simple English.

> * Prepare yourself to make a mindset that you won’t be writing this project in the future and think about the next person. Your naming will provide an insight into the process and project.

Reviewer: ```suggestion
* You might not be the only person working on your particular project in the future. Think about the next person. Your naming will provide an insight into the process and project.
```

> * Just try to avoid generic names. Use concrete / specific names, longer names if you need to, attach important details (pack much information) and mainly focus on code readability.

Reviewer 2: Except when naming components 👀  🙈

Author: removed
